test/integration/verify/controls/azurerm_ad_users.rb

```diff
@@ -0,0 +1,9 @@
+control 'azure_ad_users' do
+
+  title 'Ensure no Guest Users are present within an Active Directory Tenant'
+
+  describe azurerm_ad_users.guest_accounts do
+    it { should_not exist }
```

Comment:
Is guest_accounts `nil` or `[]` (empty list)?

Reply:
It will return an object of type GuestAccount, with an `exists` method. I figured this would read more in the style of inspec.

Reply:
I'm not sure what the output would look like if it does exist. Would you get a list of users that are guest accounts? I'm just wondering how you'd know which accounts are guest accounts so you could fix the finding.

Reply:
Output on fail would look like:
```
×  azure_ad_users: Ensure no Guest Users are present within an Active Directory Tenant
     ×  [
       {
         "displayName": "User One",
         "mail": "one@email.com",
         "userType": "Guest"
       },
       {
         "displayName": "User Two",
         "mail": "two@email.com",
         "userType": "Guest"
       }
     ] should not exist
     expected #<#<Class:0x00007f8eb52a8028>::AzurermAdUsers::GuestUsers:0x00007f8eb99fb6f0 @guests=[{ ... }], @exists=true> not to exist
```
Giving you enough information for remediation.
If no Guest users are active, exists will be false and the control will pass.